src/deepqmc/sampling.py

```diff
@@ -384,9 +418,10 @@ def rand_from_mol(mol, bs, pop_charges=None, elec_std=1.0):
     idxs = torch.repeat_interleave(
         torch.arange(n_atoms, device=cs.device).expand(bs, -1), repeats.flatten()
     ).view(bs, n_electrons)
-    idxs = torch.stack([idxs[i, torch.randperm(idxs.shape[-1])] for i in range(bs)])
+    idxs = sort_nucleus_indices(idxs, mol)
     centers = mol.coords[idxs]
     rs = centers + elec_std * torch.randn_like(centers)
+    # permute within same spin electrons
```

Comment:
this looks like a leftover TODO comment? otherwise good to go

Reply:
Thanks, yes this was an artifact that should be removed.